test/assertions/to-have-properties-satisfying.spec.js

```diff
@@ -0,0 +1,258 @@
+/* global expect */
+describe('to have properties satisfying assertion', () => {
+  it('requires a third argument', () => {
+    expect(
+      function() {
+        expect([1, 2, 3], 'to have properties satisfying');
+      },
+      'to throw',
+      'expected [ 1, 2, 3 ] to have properties satisfying\n' +
+        '  The assertion does not have a matching signature for:\n' +
+        '    <array> to have properties satisfying\n' +
+        '  did you mean:\n' +
+        '    <object> to have properties satisfying <any>\n' +
+        '    <object> to have properties satisfying <assertion>'
+    );
+  });
+
+  it('does not accept a fourth argument', () => {
+    expect(
+      function() {
+        expect([1], 'to have properties satisfying', 0, 1);
+      },
+      'to throw',
+      'expected [ 1 ] to have properties satisfying 0, 1\n' +
+        '  The assertion does not have a matching signature for:\n' +
+        '    <array> to have properties satisfying <number> <number>\n' +
+        '  did you mean:\n' +
+        '    <object> to have properties satisfying <any>\n' +
+        '    <object> to have properties satisfying <assertion>'
+    );
+  });
+
+  it('only accepts objects as the target', () => {
+    expect(
+      function() {
+        expect(42, 'to have properties satisfying', true);
+      },
+      'to throw',
+      'expected 42 to have properties satisfying true\n' +
+        '  The assertion does not have a matching signature for:\n' +
+        '    <number> to have properties satisfying <boolean>\n' +
+        '  did you mean:\n' +
+        '    <object> to have properties satisfying <any>\n' +
+        '    <object> to have properties satisfying <assertion>'
+    );
+  });
+
+  it('allows assertions as the argument', () => {
+    expect(
+      { foo: 0, bar: 1, baz: 2, qux: 3 },
+      'to have properties satisfying',
+      'not to be empty'
+    );
+
+    expect(
+      { foo: 0, bar: 1, baz: 2, qux: 3 },
+      'to have properties satisfying',
+      'not to be empty'
+    );
+
+    expect(
+      { foo: 0, bar: 1, baz: 2, qux: 3 },
+      'to have properties satisfying',
+      'to match',
+      /^[a-z]{3}$/
+    );
+  });
+
+  it('allows expect.it as an argument (the wrapped function receives the property)', () => {
+    expect(
+      { ff: 0, bbbb: 1, cc: 2 },
+      'to have properties satisfying',
+      expect.it(function(property) {
+        // properties must be of even length
+        expect(property.length % 2 === 0, 'to be true');
+      })
+    );
+  });
+
+  it('ignores properties with undefined values', () => {
+    expect(
+      { foo: undefined, bar: '123' },
+      'to have properties satisfying',
+      'to be a string'
```

Comment:
This doesn’t really test what it says it does — `foo` also satisfies theassertion.